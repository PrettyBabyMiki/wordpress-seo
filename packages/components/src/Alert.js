/* External dependencies */
import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import { __ } from "@wordpress/i18n";

/* Yoast dependencies */
import { getDirectionalStyle } from "@yoast/helpers";
import { colors } from "@yoast/style-guide";

/* Internal dependencies */
import Button from "./buttons/Button";
import SvgIcon from "./SvgIcon";

const AlertContainer = styled.div`
	display: flex;
	align-items: flex-start;
	fontSize: 14px;
	border: 1px solid rgba(0, 0, 0, 0.2);
	padding: 16px;
	color: ${ props => props.alertColor };
	background: ${ props => props.alertBackground };
	margin-bottom: 20px;
`;

const AlertContent = styled.div`
	flex-grow: 1;

	a {
		color: ${ colors.$color_alert_link_text };
	}
`;

const AlertIcon = styled( SvgIcon )`
	margin-top: 0.125rem;
	${ getDirectionalStyle( "margin-right: 8px", "margin-left: 8px" ) };
`;

const AlertDismiss = styled( Button )`
	${ getDirectionalStyle( "margin-left: 8px", "margin-right: 8px" ) };
	font-size: 24px;
	line-height: 24px;
	color: ${ props => props.alertDismissColor };

	// Override the base button style: get rid of the button styling.
	min-height: auto;
	padding: 0;

	&, &:hover, &:focus, &:active {
		border-radius: 0;
		border: none;
		background: transparent;
		box-shadow: none;
	}
`;

/**
 * Alert component.
 *
 * @param {Object}   props               The props to use.
 * @param {*}        props.children      The children to render inside the alert.
 * @param {string}   props.type          The type of Alert. Can be: "error", "info", "success" or "warning".
 *                                       Controls the colors and icon of the Alert.
 * @param {Function} [props.onDismissed] When supplied this Alert will gain an 'X' button.
 *                                       Note: the function provided should do the actual dismissing!
 *
 * @returns {Alert} The Alert component.
 */
class Alert extends React.Component {
	/**
	 * Returns the colors and icon to be used based on the type provided to the props.
	 *
	 * @param {string} type The type of Alert.
	 *
	 * @returns {object} Options with colors and icons to be used.
	 */
	getTypeDisplayOptions( type ) {
		switch ( type ) {
			case "error":
				return {
					color: colors.$color_alert_error_text,
					background: colors.$color_alert_error_background,
					icon: "alert-error",
				};
			case "info":
				return {
					color: colors.$color_alert_info_text,
					background: colors.$color_alert_info_background,
					icon: "alert-info",
				};
			case "success":
				return {
					color: colors.$color_alert_success_text,
					background: colors.$color_alert_success_background,
					icon: "alert-success",
				};
			case "warning":
				return {
					color: colors.$color_alert_warning_text,
					background: colors.$color_alert_warning_background,
					icon: "alert-warning",
				};
		}
	}

	/**
	 * Renders the component.
	 *
	 * @returns {React.Element} The rendered component.
	 */
	render() {
		const options = this.getTypeDisplayOptions( this.props.type );

		return <AlertContainer alertColor={ options.color } alertBackground={ options.background }>
			<AlertIcon icon={ options.icon } color={ options.color } />
			<AlertContent>{ this.props.children }</AlertContent>
			{
				typeof this.props.onDismissed === "function"
					? (
						<AlertDismiss
							alertDismissColor={ options.color }
							onClick={ this.props.onDismissed }
							aria-label={ __( this.props.dismissAriaLabel, "yoast-components" ) }
						>
							&times;
						</AlertDismiss>
					)
					: null
			}
		</AlertContainer>;
	}
}

Alert.propTypes = {
	children: PropTypes.any.isRequired,
	type: PropTypes.oneOf( [ "error", "info", "success", "warning" ] ).isRequired,
	onDismissed: PropTypes.func,
	dismissAriaLabel: PropTypes.string,
};

Alert.defaultProps = {
	onDismissed: null,
	dismissAriaLabel: "Dismiss this alert",
};

export default Alert;
