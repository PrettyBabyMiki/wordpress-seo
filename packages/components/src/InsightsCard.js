import React, { Fragment } from "react";
import PropTypes from "prop-types";
import HelpIcon from "./help-icon/HelpIcon";

class InsightsCard extends React.Component {

	getDescription() {
		if ( this.props.description ) {
			return this.props.description;
		}

		return;
	}

	getHelpIcon() {
		if ( this.props.linkTo ) {
			return (
				<HelpIcon
					linkTo={ this.props.linkTo }
					linkText={ this.props.linkText }
				/>
			);
		}

		return;
	}

	render() {
		return (
			<div className="yoast-insights-card">
				<div className="yoast-field-group__title">
					<b>{ this.props.title }</b>
					{ this.getHelpIcon() }
				</div>
				<div className="yoast-insights-card__content">
					<p className="yoast-insights-card__score"><span className="yoast-insights-card__amount">{ this.props.amount }</span> { this.props.measurement }</p>
					<div className="yoast-insights-card__description">{ this.getDescription() }</div>
				</div>
			</div>
		);
	}
}

InsightsCard.propTypes = {
	title: PropTypes.string,
	amount: PropTypes.number,
	description: PropTypes.element,
	measurement: PropTypes.string,
	linkTo: PropTypes.string,
	linkText: PropTypes.string,
}

export default InsightsCard;