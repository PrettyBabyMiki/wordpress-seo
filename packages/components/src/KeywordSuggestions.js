// External dependencies.
import React from "react";
import PropTypes from "prop-types";
import interpolateComponents from "interpolate-components";
import { __, sprintf } from "@wordpress/i18n";

// Yoast dependencies.
import WordList from "./WordList";

/**
 * @summary Translates and returns the keyword research article link.
 *
 * Translates and returns the keyword research article link
 * in a way that the link elements are still rendered.
 *
 * @returns {JSX.Element} The translated text including rendered link components.
 */
const getKeywordResearchArticleLink = () => {
	const keywordsResearchLinkTranslation = sprintf(
		__(
			"Read our %1$sultimate guide to keyword research%2$s to learn " +
			"more about keyword research and keyword strategy.",
			"yoast-components"
		),
		"{{a}}",
		"{{/a}}"
	);

	return interpolateComponents( {
		mixedString: keywordsResearchLinkTranslation,
		components: {
			// eslint-disable-next-line jsx-a11y/anchor-has-content
			a: <a href="https://yoa.st/keyword-research-metabox" />,
		},
	} );
};

/**
 * @summary Determine the keyword suggestion explanation.
 *
 * @param {Array} keywords The keyword suggestions that were found.
 * @returns {string} The translated text.
 */
const getKeywordSuggestionExplanation = keywords => {
	if ( keywords.length === 0 ) {
		return __(
			"Once you add a bit more copy, we'll give you a list of words and " +
			"word combination that occur the most in the content. These give an indication of what your content focuses on.",
			"yoast-components"
		);
	}

	return __(
		"The following words and word combinations occur the most in the content. " +
		"These give an indication of what your content focuses on. " +
		"If the words differ a lot from your topic, " +
		"you might want to rewrite your content accordingly. ",
		"yoast-components"
	);
};

/**
 * @summary WordList component.
 *
 * @param {string}   title           The title of the list.
 * @param {string}   relevantWords   The relevant words.
 * @param {number}   keywordLimit    The maximum number of keywords to display.
 *
 * @returns {JSX.Element} Rendered WordList component.
 */
const KeywordSuggestions = ( { relevantWords, keywordLimit } ) => {
	const keywords = relevantWords.slice( 0, keywordLimit ).map( word => word.getCombination() );

	return ( <WordList
		title={ __( "Prominent words", "yoast-components" ) }
		words={ keywords }
		classNamePrefix="yoast-keyword-suggestions"
		showBeforeList={ () => {
			return  ( <p>{ getKeywordSuggestionExplanation( keywords ) }</p> );
		} }
		showAfterList={
			() => {
				return getKeywordResearchArticleLink();
			}
		}
	/>
	);
};

KeywordSuggestions.propTypes = {
	relevantWords: PropTypes.arrayOf( PropTypes.object ).isRequired,
	keywordLimit: PropTypes.number,
};

KeywordSuggestions.defaultProps = {
	keywordLimit: 5,
};

export default KeywordSuggestions;
