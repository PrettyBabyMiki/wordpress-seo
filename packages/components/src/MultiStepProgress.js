// External dependencies.
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

// Yoast dependencies.
import { colors } from "@yoast/style-guide";

import SvgIcon from "./SvgIcon";

/**
 * Ordered list containing the steps.
 */
const MultiStepProgressContainer = styled.ol`
	list-style: none;
	padding-left: 2px;
	counter-reset: multi-step-progress-counter;
	
	background: white;

	li {
		counter-increment: multi-step-progress-counter;
	}
`;

/**
 * List item for a single step.
 */
const MultiStepProgressListItem = styled.li`
	margin: 4px 0;
	padding: 4px;
	display: flex;
	flex-direction: row;
	align-items: baseline;
	
	span {
		margin: 0 8px;
	}
	
	svg {
		position: relative;
		top: 2px;
		margin-right: 8px;
	}
	
	::before {
		content: counter( multi-step-progress-counter );
		font-size: 12px;
		background: ${ colors.$color_pink_dark };
		border-radius: 50%;
		min-width: 16px;
		height: 16px;
		padding: 4px;
		color: ${ colors.$color_white };
		text-align: center;
		display: inline-block;
	}
`;

/**
 * Specific list item for a step in pending state.
 */
const MultiStepProgressPendingStatus = styled( MultiStepProgressListItem )`
	span {
		color: ${ colors.$palette_grey_text_light };
	}
	
	::before {
		background-color: ${ colors.$palette_grey_medium_dark };
	}
`;

/**
 * Specific list item for a step in running state.
 */
const MultiStepProgressRunningStatus = styled( MultiStepProgressListItem )`
	::before {
		background-color: ${ colors.$palette_grey_medium_dark };
	}
`;

/**
 * A component to track multiple steps in a process.
 */
class MultiStepProgress extends React.Component {
	/**
	 * Returns the rendered html.
	 *
	 * @returns {React.Element} The rendered html.
	 */
	render() {
		return (
			<MultiStepProgressContainer>
				{ this.props.steps.map( ( step ) => {
					switch ( step.status ) {
						case "running":
							return this.renderRunningState( step );
						case "failed":
							return this.renderFailedState( step );
						case "finished":
							return this.renderFinishedState( step );
						case "pending":
						default:
							return this.renderPendingState( step );
					}
				} ) }
			</MultiStepProgressContainer>
		);
	}

	/**
	 * Renders a list item in pending state.
	 *
	 * @param {Object} step Step content.
	 *
	 * @returns {React.Element} The list item.
	 */
	renderPendingState( step ) {
		return (
			<MultiStepProgressPendingStatus key={ step.id }>
				<span>{ step.text }</span>
			</MultiStepProgressPendingStatus>
		);
	}

	/**
	 * Renders a list item in running state.
	 *
	 * @param {Object} step Step content.
	 *
	 * @returns {React.Element} The list item.
	 */
	renderRunningState( step ) {
		return (
			<MultiStepProgressRunningStatus key={ step.id }>
				<span>{ step.text }</span>
				<SvgIcon icon="loading-spinner" />
			</MultiStepProgressRunningStatus>
		);
	}

	/**
	 * Renders a list item in finished state.
	 *
	 * @param {Object} step Step content.
	 *
	 * @returns {React.Element} The list item.
	 */
	renderFinishedState( step ) {
		return (
			<MultiStepProgressListItem key={ step.id }>
				<span>{ step.text }</span>
				<SvgIcon icon="check" color={ colors.$color_green_medium_light } />
			</MultiStepProgressListItem>
		);
	}

	/**
	 * Renders a list item in failed state.
	 *
	 * @param {Object} step Step content.
	 *
	 * @returns {React.Element} The list item.
	 */
	renderFailedState( step ) {
		return (
			<MultiStepProgressListItem key={ step.id }>
				<span>{ step.text }</span>
				<SvgIcon icon="times" color={ colors.$color_red } />
			</MultiStepProgressListItem>
		);
	}
}

MultiStepProgress.defaultProps = {
	steps: [],
};

MultiStepProgress.propTypes = {
	steps: PropTypes.arrayOf( PropTypes.shape( {
		status: PropTypes.oneOf( [ "pending", "running", "finished", "failed" ] ).isRequired,
		text: PropTypes.string.isRequired,
		id: PropTypes.string.isRequired,
	} ) ),
};

export default MultiStepProgress;
