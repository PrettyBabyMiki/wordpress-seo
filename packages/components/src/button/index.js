import "./buttons.css";

export { Button, ButtonStyledLink } from "./Button.js";
export { CloseButton } from "./Icon.js";
