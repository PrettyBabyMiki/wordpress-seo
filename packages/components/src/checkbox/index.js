import "./checkbox.css";

export { default as CheckboxGroup } from "./CheckboxGroup";
