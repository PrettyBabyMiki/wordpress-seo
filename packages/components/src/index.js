import StyledSection, { StyledSectionBase, StyledHeading } from "./StyledSection";

export {
	StyledSection,
	StyledSectionBase,
	StyledHeading,
};

export { icons } from "./SvgIcon";
export { default as Button } from "./Button";
export { default as ErrorBoundary } from "./ErrorBoundary";
export { default as Explanation } from "./Explanation";
export { default as Heading } from "./Heading";
export { default as IFrame } from "./IFrame";
export { default as Input } from "./Input";
export { default as Label } from "./Label";
export { default as Loader } from "./Loader";
export { default as Notification } from "./Notification";
export { default as Paper } from "./Paper";
export { default as Progressbar } from "./Progressbar";
export { default as Section } from "./Section";
export { default as SvgIcon } from "./SvgIcon";
export { default as Textarea } from "./Textarea";
export { default as Textfield } from "./Textfield";
export { default as YoastLogo } from "./YoastLogo";
export { default as YoastSeoIcon } from "./YoastSeoIcon";
export { default as YouTubeVideo } from "./YouTubeVideo";

export { ListTable, ZebrafiedListTable } from "./table/ListTable";
export { Row } from "./table/Row";
