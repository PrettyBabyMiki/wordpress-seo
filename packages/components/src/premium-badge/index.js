import "./premium-badge.css";
export { default as PremiumBadge } from "./PremiumBadge";
