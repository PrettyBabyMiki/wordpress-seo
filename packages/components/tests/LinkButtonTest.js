import React from "react";
import renderer from "react-test-renderer";

import { BaseLinkButton, LinkButton } from "../src/index";

test( "the BaseLinkButton matches the snapshot", () => {
	const component = renderer.create(
		<BaseLinkButton>LinkButtonValue</BaseLinkButton>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the LinkButton matches the snapshot", () => {
	const component = renderer.create(
		<LinkButton>LinkButtonValue</LinkButton>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
