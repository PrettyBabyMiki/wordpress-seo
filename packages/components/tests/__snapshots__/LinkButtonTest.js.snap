// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the LinkButton matches the snapshot 1`] = `
.c6:active {
  color: #000;
  background-color: #f7f7f7;
  border-color: #888;
  box-shadow: inset 0 2px 5px -3px rgba( 136,136,136,0.5 );
}

yoast-seo.g4[id="c1"] {
  content: "c6,";
}

.c7::-moz-focus-inner {
  border-width: 0;
}

.c7:focus {
  outline: none;
  border-color: #0066cd;
  color: #000;
  background-color: #fff;
  box-shadow: 0 0 3px rgba( 8,74,103,0.8 );
}

yoast-seo.g5[id="c2"] {
  content: "c7,";
}

.c8:hover {
  color: #000;
  background-color: #fff;
  border-color: var(--yoast-color-border--default);
}

yoast-seo.g6[id="c3"] {
  content: "c8,";
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 4px 10px;
  border-radius: 3px;
  cursor: pointer;
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  font-weight: inherit;
  text-align: left;
  overflow: visible;
  min-height: 32px;
  -webkit-transition: var(--yoast-transition-default);
  transition: var(--yoast-transition-default);
}

.c9 svg {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

yoast-seo.g7[id="c4"] {
  content: "c9,";
}

.c5 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #555;
  border-color: #ccc;
  background: #f7f7f7;
  box-shadow: 0 1px 0 rgba( 204,204,204,1 );
}

yoast-seo.g101[id="c0"] {
  content: "c5,";
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c9::after {
    display: inline-block;
    content: "";
    min-height: 22px;
  }
}

<a
  className="c0 c1 c2 c3 c4 c5 c6 c7 c8 c9"
>
  LinkButtonValue
</a>
`;
